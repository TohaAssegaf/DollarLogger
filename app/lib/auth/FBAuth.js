import { AccessToken, LoginManager } from 'react-native-fbsdk'
import firebase from 'react-native-firebase'

export const login = (error, result) => {
  if (error) {
    throw new Error('Login failed with error: ' + error.message)
  }
  if (result.isCancelled) {
    throw new Error('User cancelled request.')
  }
  return AccessToken.getCurrentAccessToken().then(data => {
    if (!data) {
      throw new Error('Access token could not be obtained.')
    }
    const credential = firebase.auth.FacebookAuthProvider.credential(data.accessToken)
    return firebase.auth().signInAndRetrieveDataWithCredential(credential)
  })
}
