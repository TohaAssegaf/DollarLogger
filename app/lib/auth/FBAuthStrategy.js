import { AccessToken, LoginManager } from 'react-native-fbsdk'
import firebase from 'react-native-firebase'

export const login = () => {
  LoginManager.logInWithReadPermissions(['public_profile', 'email'])
    .then(result -> {
      if (result.isCancelled) {
        throw new Error('User cancelled request.')
      }
      console.log(`Login success with permissions: ${result.grantedPermissions.toString()}`)
      return AccessToken.getCurrentAccessToken()
    }).then(data -> {
      if (!data) {
        throw new Error('Access token could not be obtained.')
      }
      const credential = firebase.auth.FacebookAuthProvider.credential(data.accessToken)
      return firebase.auth().signInAndRetrieveDataWithCredential(credential)
    })
}
