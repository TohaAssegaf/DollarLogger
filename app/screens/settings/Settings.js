import styles from './styles'
import actions from '~/app/actions'
import SettingsCell from '~/app/components/settings/SettingsCell'
import { HEADER_BACKGROUND_COLOR, HEADER_TEXT_COLOR} from '~/app/config/colors'
import * as Routes from '~/app/config/Routes'
import React from 'react';
import { Text, View } from 'react-native'

export default class Settings extends React.Component {
  static navigationOptions = {
    title: 'Settings',
    headerStyle: {
      backgroundColor: HEADER_BACKGROUND_COLOR,
    },
    headerTintColor: HEADER_TEXT_COLOR
  }

  navigateToUpdateBudget() {
    this.props.navigation.navigate(Routes.UPDATE_BUDGET, { isUpdateExistingBudget: true })
  }

  navigateToLogin() {
    this.props.navigation.navigate(Routes.LOGIN)
  }

  render() {
    return (
      <View style={styles.screen}>
        <SettingsCell
          text='Update budget'
          onTap={() => this.navigateToUpdateBudget()} />
          <SettingsCell
            text='Sign in to back-up your data'
            onTap={() => this.navigateToLogin()} />
      </View>
    );
  }
}
