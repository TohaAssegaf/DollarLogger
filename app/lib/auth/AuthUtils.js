import firebase from 'react-native-firebase'
import * as FBAuth from './FBAuth'

const AUTH_STRATEGIES = [FBAuth]

export function isLoggedIn() {
  return firebase.auth().currentUser !== null
}

export function logout() {
  firebase.auth().signOut()
  for (const strategy of AUTH_STRATEGIES) {
    strategy.logout()
  }
}

export function onAuthStateChanged(listener: Function) {
  return firebase.auth().onAuthStateChanged(listener)
}
