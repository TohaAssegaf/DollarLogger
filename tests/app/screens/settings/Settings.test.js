import * as AuthUtils from '~/app/lib/auth/AuthUtils'
import Settings from '~/app/screens/settings/Settings'
import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const component = <Settings />
  const loggedInRender = renderer.create(component).toJSON()
  expect(loggedInRender).toMatchSnapshot()

  AuthUtils.logout()
  const loggedOutRender = renderer.create(component).toJSON()
  expect(loggedOutRender).toMatchSnapshot()
});
