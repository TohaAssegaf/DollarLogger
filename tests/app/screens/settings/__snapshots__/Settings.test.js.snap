// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "action": [Function],
          "displayName": "Update budget",
        },
        Object {
          "action": [Function],
          "displayName": "Sign out",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "#222222",
                "borderBottomWidth": 0.5,
                "height": 70,
                "justifyContent": "center",
                "paddingHorizontal": 20,
              }
            }
          >
            <Text>
              Update budget
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "borderBottomColor": "#222222",
                "borderBottomWidth": 0.5,
                "height": 70,
                "justifyContent": "center",
                "paddingHorizontal": 20,
              }
            }
          >
            <Text>
              Sign out
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
